django/db/models/expressions.py

```diff
@@ -778,9 +786,9 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
         c.result = c.result.resolve_expression(query, allow_joins, reuse, summarize, for_save)
         return c
 
-    def as_sql(self, compiler, connection, template=None):
+    def as_sql(self, compiler, connection, template=None, **extra_context):
```

Comment:
I won't audit every case in this review, but as above, it doesn't seem that `extra_context` provides any functionality that customizing the `template` doesn't already allow.